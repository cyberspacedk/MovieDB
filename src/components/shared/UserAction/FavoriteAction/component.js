import React, { useState } from 'react';
import { Icon } from 'antd';

const FavoriteAction = () => {
  const [favoriteStatus, setFavorite] = useState(false);
  const handleFavorite = () => setFavorite(!favoriteStatus);

  return (
    <Icon
      type="heart"
      theme={favoriteStatus ? 'filled' : undefined}
      onClick={handleFavorite}
    />
  );
};

export default FavoriteAction;
