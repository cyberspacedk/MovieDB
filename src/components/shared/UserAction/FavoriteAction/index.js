import FavoriteActionContainer from './container';

export default FavoriteActionContainer;
