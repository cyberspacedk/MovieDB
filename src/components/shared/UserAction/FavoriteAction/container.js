import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import FavoriteAction from './component';
import { operationsFavoritesRequest } from '../../../../store/favorites/actions';

class FavoriteActionContainer extends Component {
  state = {
    favoriteStatus: false,
  };

  handleFavorite = () => {
    const { operationsFavoritesRequest, movieId } = this.props;
    this.setState(prevState => ({ favoriteStatus: !prevState.favoriteStatus }));
    operationsFavoritesRequest(movieId, true);
  };

  render() {
    const { favoriteStatus } = this.state;
    return (
      <FavoriteAction
        handleFavorite={this.handleFavorite}
        favoriteStatus={favoriteStatus}
      />
    );
  }
}

FavoriteActionContainer.propTypes = {
  operationsFavoritesRequest: PropTypes.func.isRequired,
  movieId: PropTypes.number.isRequired,
};

const mdtp = {
  operationsFavoritesRequest,
};

export default connect(
  null,
  mdtp,
)(FavoriteActionContainer);
