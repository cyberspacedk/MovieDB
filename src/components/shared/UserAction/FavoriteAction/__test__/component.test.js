import React from 'react';
import { shallow } from 'enzyme';
import FavoriteAction from '../component';

describe('Component: CreateListAction', () => {
  it('Match its snapshot.', () => {
    const props = {
      handleFavorite: jest.fn(),
      favoriteStatus: false,
    };

    const wrapper = shallow(<FavoriteAction {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Match its snapshot.', () => {
    const props = {
      handleFavorite: jest.fn(),
      favoriteStatus: true,
    };
    const wrapper = shallow(<FavoriteAction {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
