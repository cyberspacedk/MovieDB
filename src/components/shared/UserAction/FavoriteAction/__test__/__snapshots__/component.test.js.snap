// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CreateListAction Match its snapshot. 1`] = `
<Icon
  onClick={[MockFunction]}
  type="heart"
/>
`;

exports[`Component: CreateListAction Match its snapshot. 2`] = `
<Icon
  onClick={[MockFunction]}
  theme="filled"
  type="heart"
/>
`;
